src/main/java/com/ss/ucm/ms/loans/controllers/LoansController.java

```diff
@@ -34,47 +48,162 @@
-    @Autowired
-    UserLoanAdd userLoanAdd;
-    
-    
-    @Autowired
-    public LoansController(LoanSearch loanSearch) {
-        this.loanSearch = loanSearch;
-    }
-
-    /**
-     * GET /api/loans - Return all loans
-     * @return First 50 loans
-     */
-    @GetMapping
-    public ResponseEntity<Collection<Loan>> get() {
-        try {
-            Collection<Loan> loans = loanSearch.getFirst50();
-            return new ResponseEntity<>(loans, HttpStatus.OK);
-        } catch (Exception e) {
-            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
-        }
-    }
-    
-    
 	/**
 	 * 
-	 * @param userId, loanRequest(body)
-	 * @return void
+	 * @param salary(user), amount of loan, loan Term, loan interest rate.
+	 * @return ResponseEntity
 	 * 
 	 */
-     
+	
 	@PostMapping("/loansignup")
-	public void signupLoan(@RequestParam int userId,  @RequestBody RequestLoanSignupDto loanRequest) {
-			
-		loanAdd.signUpLoan(userId, loanRequest);
+	public ResponseEntity<?>signupLoan(@RequestParam int salary, @RequestParam int amount, @RequestParam int term, @RequestParam double interestRate) {
+
+		int termInMonths = term * 12;
+
+		ResponseLoanMonthlyPaymentDto loanDto = new ResponseLoanMonthlyPaymentDto();
+
+
+		double monthlyPayments = loanAdd.calculateMonthlyPayments(amount, term, interestRate);
+
+		double totalPayments = loanAdd.calculateTotalPayment(monthlyPayments, termInMonths);
+
+		loanDto.setPayments(monthlyPayments);
+		loanDto.setTotalPayment(totalPayments);
+
+		return  ResponseEntity.ok((new ResponseLoanMonthlyPaymentDto(loanDto.getPayments(), 
+				loanDto.getTotalPayment())));
+
+	}
+
+	
+	
+	/**
+	 * 
+	 * @param user id, loanrequest
+	 * @return void
+	 * 
+	 */
+
+	@PostMapping("/loanSignupSuccess")
+	public void signupLoanSuccess(@RequestParam int userId,  @RequestBody RequestLoanSignupDto loanRequest) {
+
+		//Before setting is_accpeted to true/loan accepeted... we need to 
+		// make sure that our customer makes atleast 50% of what their loan amount is.
+		double acceptableIncome = loanRequest.getBalance().doubleValue() * 0.50;
 		
+		User toEmail = userDao.getEmailById(userId);
+
+		String to = toEmail.getEmail();
+		String from = "utopiacashmoney99@gmail.com";
+		final String username = "utopiacashmoney99@gmail.com";
+		final String password = "UtopiaBanking100?";
+
+		String host = "smtp.gmail.com";
+
+		Properties props = new Properties();
+		props.put("mail.smtp.auth", "true");
+		props.put("mail.smtp.starttls.enable", "true");
+		props.put("mail.smtp.host", host);
+		props.put("mail.smtp.port", "587");
+
+
+		Session session = Session.getInstance(props,
+				new javax.mail.Authenticator() {
+			protected PasswordAuthentication getPasswordAuthentication() {
+				return new PasswordAuthentication(username, password);
+			}
+		});
+
+		Message message = new MimeMessage(session);
+
+
+
+
+		if(loanRequest.getSalary() >= acceptableIncome) {
+			
+			loanAdd.signUpLoan(userId, loanRequest);
```

Comment:
Input validation exists for the front end, but not the back end - don't trust user input!

Reply:
I fixed the validation in the front end which won't even allow users to enter anything but numbers and not just "0"